### PR TITLE
[SYCL][XRT][Vitis] Fix some failing tests + improve device detection of xrt_pi

### PR DESCRIPTION
**Pull request by Ralender**

### CODE REVIEW

File: sycl/plugins/xrt/pi_xrt.cpp [hunk 1]
```diff
@@ -339,7 +339,9 @@ struct _pi_device
 
 private:
   native_type xrtDevice_;
-  ref_counted_ref<_pi_platform> platform_;
+  /// _pi_platform hold a counting reference onto all its _pi_device so we do
```
Comment (keryell) [inline sycl/plugins/xrt/pi_xrt.cpp]:
```suggestion
  /// _pi_platform holds a counting reference onto all its _pi_device so we do
```

File: sycl/plugins/xrt/pi_xrt.cpp [hunk 2]
```diff
@@ -360,16 +362,23 @@ struct _pi_platform : ref_counted_base<_pi_platform>, unique<_pi_platform> {
 private:
   /// _pi_device hold a counting reference onto _pi_platform so we do not keep
   /// counting reference on devices to prevent circular dependency.
-  std::vector<_pi_device*> devices_;
+  std::vector<ref_counted_ref<_pi_device>> devices_;
```
Comment (keryell) [inline sycl/plugins/xrt/pi_xrt.cpp]: I guess you need to update the comment accordingly.

**Review by keryell — CHANGES_REQUESTED** [review]
Thanks.
You can also fix the type in the commit message.

**Comment by Ralender** [comment]
do not merge yet I started merging from upstream and this will be rebased on the merge

**Review by keryell — CHANGES_REQUESTED** [review]
Looks good except the typo in the commit message: `failling` → `failling`.
But since you have to rebase this, you can edit the message at the same time.

**Review by keryell — APPROVED** [review]
Thanks!

**Pull request merged**